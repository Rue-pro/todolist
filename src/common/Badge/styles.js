import styled, { css } from 'styled-components'
import { colors } from '../constants'

export const BadgeStyled = styled.div`
  box-sizing: border-box;
  min-width: 130px;
  padding: 12px 18px;
  border-radius: 23px;
  font-weight: 500;
  ${(props) => {
    const { color, bgColor } = props
    return css`
      background-color: ${bgColor};
      color: ${color || `rgb(${colors.gray})`};
    `
  }}
`
