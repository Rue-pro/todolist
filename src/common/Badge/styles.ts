import styled from 'styled-components'
import { colors } from '../constants'

interface IBadgeStyledProps {
  color: string
  bgColor: string
}

export const BadgeStyled = styled.div<IBadgeStyledProps>`
  display: flex;
  justify-content: center;
  box-sizing: border-box;
  min-width: 130px;
  padding: 12px 18px;
  border-radius: 23px;
  font-weight: 500;
  background-color: ${(props) => props.bgColor};
  color: ${(props) => props.color || `rgb(${colors.gray})`};
`
