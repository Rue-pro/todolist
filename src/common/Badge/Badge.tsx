import React from 'react'
import { colors } from '../constants'
import { Text } from '../Text/Text'
import { BadgeStyled } from './styles'

export const Badge = (props: any) => {
  let color = 'black'
  let bgColor = 'black'

  switch (props.children) {
    case 'Approved':
      color = colors.primary
      bgColor = colors.purple_light
      break
    case 'In Progress':
      color = colors.green
      bgColor = colors.green_light
      break
    case 'In Review':
      color = colors.red
      bgColor = colors.red_light
      break
    case 'Waiting':
      color = colors.gray
      bgColor = colors.gray_light
      break
  }

  return (
    (props.children && (
      <BadgeStyled bgColor={bgColor} color={color} {...props}>
        <Text color={color}>{props.children}</Text>
      </BadgeStyled>
    )) || <></>
  )
}
