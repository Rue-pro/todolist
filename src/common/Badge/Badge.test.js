import React from 'react'
import Badge from './Badge'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { BadgeTypeEnum } from './Badge'

describe('Badge', () => {
  it('should render correctly with type=approved', () => {
    const output = shallow(<Badge type={BadgeTypeEnum.approved}>approved badge</Badge>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with type=inProgress', () => {
    const output = shallow(<Badge type={BadgeTypeEnum.inProgress}>inProgress badge</Badge>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with type=inReview', () => {
    const output = shallow(<Badge type={BadgeTypeEnum.inReview}>inReview badge</Badge>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with type=waiting', () => {
    const output = shallow(<Badge type={BadgeTypeEnum.waiting}>waiting badge</Badge>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with wrong props type', () => {
    const output = shallow(<Badge type="approved">wrong type badge</Badge>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with no props children', () => {
    const output = shallow(<Badge type={BadgeTypeEnum.approved}></Badge>)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
  it('should render correctly with no props', () => {
    const output = shallow(<Badge />)
    expect(shallowToJson(output)).toMatchSnapshot()
  })
})
