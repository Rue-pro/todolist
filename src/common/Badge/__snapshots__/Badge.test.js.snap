// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render correctly with no props 1`] = `<Fragment />`;

exports[`Badge should render correctly with no props children 1`] = `<Fragment />`;

exports[`Badge should render correctly with type=approved 1`] = `
<styled.div
  bgColor="#ECE7FE"
  color="#623CE7"
>
  <Text
    color="#623CE7"
  >
    approved badge
  </Text>
</styled.div>
`;

exports[`Badge should render correctly with type=inProgress 1`] = `
<styled.div
  bgColor="#E1FAF4"
  color="#31B195"
>
  <Text
    color="#31B195"
  >
    inProgress badge
  </Text>
</styled.div>
`;

exports[`Badge should render correctly with type=inReview 1`] = `
<styled.div
  bgColor="#FEE8EA"
  color="#D83749"
>
  <Text
    color="#D83749"
  >
    inReview badge
  </Text>
</styled.div>
`;

exports[`Badge should render correctly with type=waiting 1`] = `
<styled.div
  bgColor="#F7F8FC"
  color="#9196A7"
>
  <Text
    color="#9196A7"
  >
    waiting badge
  </Text>
</styled.div>
`;

exports[`Badge should render correctly with wrong props type 1`] = `
<styled.div
  bgColor="black"
  color="black"
>
  <Text
    color="black"
  >
    wrong type badge
  </Text>
</styled.div>
`;
